lib/models/user.dart

```diff
@@ -0,0 +1,29 @@
+import './model.dart';
+
+class User implements Model {
+  final int id;
+  final String name;
```

Comment:
Did we want to link a user to a device in any capacity?

Reply:
I was picturing user as a user in a chat not the user who owns the device

Reply:
yeah but from a routing perspective to send a message to a person we need to know the address to route to.

Reply:
Can we assume that each device has one "user" at a time? So we just send messages to a device id and the current active user (name just for clarity to others with devices) can view the message attached to the referenced conversation (group id?)

Reply:
we can do private public keys for device id so that no sneaky buggers can pretend to be the correct device

Reply:
I think user is the wrong name for this model because you guys are seeing it differently to how I pictured it. I pictured users as a way to locally store the people you have conversations with. Not actually talk to them or provide communications capabilities more like name, image, description things like that

Reply:
That sounds reasonable. Then we should store device information somewhere with a private and public key. We need to decide if we are going to store this info in firmware or mobile. I have considered storing it on firmware, we have a 512kb flash with cache on our SoC, not all that memory is directly available to us as some is used by the SoC. It may be easier to do the check for proper address without routing back to mobile.